tests/resources/functional/snowflake/cte/cte_set_operation_precedence.sql

```diff
@@ -0,0 +1,17 @@
+--
+-- CTEs are visible to all the SELECT queries within a subsequent sequence of set operations.
+--
+
+-- snowflake sql:
+WITH a AS (SELECT 1, 2, 3)
+
+SELECT 4, 5, 6
+UNION
+SELECT * FROM a;
+
+-- databricks sql:
+WITH a AS (SELECT 1, 2, 3)
+
+(SELECT 4, 5, 6)
```

Comment:
I'm not sure the parenthesis are required here

Reply:
With respect to the semantics of the query you are correct in that they're not _required_. However they are harmless (and unambiguous).

With respect to the functional tests, they are currently required because the output stage produces them. (I haven't looked into the output stage, but assume parentheses are there because this is the easiest "safe" and correct thing to do: eliminating redundant parentheses falls into the category of optimisations.)

Reply:
Then shouldn't we create a ticket for it ?

Reply:
That's a good question. I've created #1281 to cover this topic.